compiler/luci/profile/src/CircleNodeOrigin.cpp

```diff
@@ -146,6 +146,9 @@ bool has_origin(const luci::CircleNode *circle_node)
 
 void add_origin(luci::CircleNode *circle_node, const std::shared_ptr<CircleNodeOrigin> origin)
 {
+  if (origin == nullptr)
+    return;
+
   circle_node->annot<CircleNodeOriginAnnotation>(nullptr);
```

Comment:
Maybe it would rarely happen but what if `add_origin(circle_node, nullptr)` is called where `circle_node` has `CircleNodeOriginAnnotation`. Then, 
1. remove existing notation and doing nothing.
2. just doing nothing.

Your code has chosen number 2. How about number 2?

Reply:
I think it's about the `what` or the `purpose` the method does... this is what I think
- `add` will add `origin`, which should not be an empty: thus we should drop an error if this condition does not hold.
- additionaly there will be a `clear` method that just clears the annotation.
- it is up to the caller how to use these APIs.

that is, a method should be clear how it works: we usually use `assert` to check the conditions to be clear.

Reply:
@seanshpark Actually, it's a bit complicated. 

When we first designed the origin, we assumed that origin is optional because it is just for profiling. But anyway we thought all the nodes(operators) from the loaded model would have its origin because it would be assigned in the import step. So, all profile-related method assumed the parameter(origin ptr) won't be `nullptr`.

But, unfortunately, each pass test script failed then. Because there's no origin assignment in the pass test. So, we came up with two solutions.
1. add origin assignment step into every single tests. 
2. change the API design to another where the parameter can be `nullptr`.

We've chosen the latter because we thought that nodes with no origin data can be run at the passes because origin is optional. With the latter, it is natural that `add_origin` with `nullptr` is doing nothing. I think API spec as a comment better to let developers know about the parameter being able to `nullptr`.

Reply:
> I think API spec as a comment better to let developers know about the parameter being able to nullptr.

This seems necessary

Reply:
Maybe... I should call the function as `set_origin` :(
If the name is `set_origin`, both `set_origin(node, nullptr)` and `set_origin(node, origin1)` make sense...